### PR TITLE
Greatly improved performance of replication selects in mysql

### PR DESCRIPTION
**Pull request by mlaflamm**
Now use proper indexes...

### CODE REVIEW

File: mry-core/src/main/scala/com/wajam/mry/storage/mysql/MysqlStorage.scala [hunk 1]
```diff
@@ -29,6 +29,9 @@ class MysqlStorage(config: MysqlStorageConfiguration, garbageCollection: Boolean
   private var openReadIterators: List[MutationGroupIterator] = List()
```
Comment (mlaflamm) [inline mry-core/src/main/scala/com/wajam/mry/storage/mysql/MysqlStorage.scala:29]: Changes in this file are only new timers and new logging.

File: mry-core/src/main/scala/com/wajam/mry/storage/mysql/MysqlStorage.scala [hunk 2]
```diff
@@ -544,9 +557,12 @@ class MysqlStorage(config: MysqlStorageConfiguration, garbageCollection: Boolean
...
         val grouped: mutable.Map[Timestamp, MutationGroup] = mutable.Map()
-        for (table <- tables; record <- trx.getTransactionRecords(table, from, to, ranges)) {
-          val group = grouped.getOrElseUpdate(record.timestamp, MutationGroup(record.token, record.timestamp))
-          group.records = record :: group.records
+        for (table <- tables) {
+          trace("loadTablesData: {}", table)
+          for (record <- trx.getTransactionRecords(table, from, to, ranges)) {
+            val group = grouped.getOrElseUpdate(record.timestamp, MutationGroup(record.token, record.timestamp))
+            group.records = record :: group.records
+          }
```
Comment (alexbergeron) [inline mry-core/src/main/scala/com/wajam/mry/storage/mysql/MysqlStorage.scala:565]: If you want to do this faster

``` scala
tables foreach (table =>  {
  trace("loadTablesData: {}", table)
  trx.getTransactionRecords(table, from, to, ranges)) foreach (record => {
    val group = grouped.getOrElseUpdate(record.timestamp, MutationGroup(record.token, record.timestamp))
    group.records = record :: group.records
  })
})
```

Comment (mlaflamm) [inline mry-core/src/main/scala/com/wajam/mry/storage/mysql/MysqlStorage.scala:565]: The optimisation is not in this file Alex! :stuck_out_tongue_winking_eye: I've just split the for in 2 to add a trace...

**Comment by coderunner** [comment]
+1

**Pull request merged**